samples/bluetooth/connection_event_trigger/src/main.c

```diff
@@ -0,0 +1,379 @@
+
+		if (conn == active_conn) {
+			bt_conn_unref(active_conn);
+			active_conn = NULL;
+		}
+
+		return;
+	}
+
+	printk("Connected: %s\n", addr);
+
+	active_conn = bt_conn_ref(conn);
+	connection_established = true;
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Disconnected: %s (reason %u)\n", addr, reason);
+
+	if (conn == active_conn) {
+		bt_conn_unref(active_conn);
+		active_conn = NULL;
+	}
+}
+
+static void le_param_updated(struct bt_conn *conn, uint16_t interval, uint16_t latency,
+			     uint16_t timeout)
+{
+	if (conn == active_conn) {
+		printk("Connection parameters updated. New interval: 1.25 * %u ms\n", interval);
+	}
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+	.le_param_updated = le_param_updated,
+};
+
+static void adv_start(void)
+{
+	int err;
+
+	err = bt_le_adv_start(BT_LE_ADV_CONN, ad, ARRAY_SIZE(ad), sd, ARRAY_SIZE(sd));
+	if (err) {
+		printk("Advertising failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Advertising successfully started\n");
+}
+
+static void scan_filter_match(struct bt_scan_device_info *device_info,
+			      struct bt_scan_filter_match *filter_match, bool connectable)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(device_info->recv_info->addr, addr, sizeof(addr));
+
+	printk("Filters matched. Address: %s connectable: %d\n", addr, connectable);
+}
+
+static void scan_connecting_error(struct bt_scan_device_info *device_info)
+{
+	printk("Connecting failed\n");
+}
+
+BT_SCAN_CB_INIT(scan_cb, scan_filter_match, NULL, scan_connecting_error, NULL);
+
+static void scan_start(void)
+{
+	int err;
+
+	struct bt_le_scan_param scan_param = {
+		.type = BT_LE_SCAN_TYPE_PASSIVE,
+		.options = BT_LE_SCAN_OPT_FILTER_DUPLICATE,
+		.interval = 0x0010,
+		.window = 0x0010,
+	};
+
+	struct bt_le_conn_param *conn_param =
+		BT_LE_CONN_PARAM(INTERVAL_100MS, INTERVAL_100MS, 0, 400);
+
+	struct bt_scan_init_param scan_init = {
+		.connect_if_match = true,
+		.scan_param = &scan_param,
+		.conn_param = conn_param,
+	};
+
+	bt_scan_init(&scan_init);
+	bt_scan_cb_register(&scan_cb);
+
+	err = bt_scan_filter_add(BT_SCAN_FILTER_TYPE_UUID,
+				 BT_UUID_DECLARE_128(ADVERTISING_UUID128));
+
+	if (err) {
+		printk("Scanning filters cannot be set (err %d)\n", err);
+		return;
+	}
+
+	err = bt_scan_filter_enable(BT_SCAN_UUID_FILTER, false);
+	if (err) {
+		printk("Filters cannot be turned on (err %d)\n", err);
+	}
+
+	err = bt_scan_start(BT_SCAN_TYPE_SCAN_PASSIVE);
+	if (err) {
+		printk("Starting scanning failed (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning successfully started\n");
+}
+
+int main(void)
+{
+	int err;
+
+	k_work_init(&work, work_handler);
+
+	IRQ_DIRECT_CONNECT(SWI_IRQn, 5, egu0_handler, 0);
+	nrf_egu_int_enable(NRF_EGU0, NRF_EGU_INT_TRIGGERED0);
+	NVIC_EnableIRQ(SWI_IRQn);
```

Comment:
I don't think it makes sense to do this here.
Could you move it to the setup function instead? Maybe next to the PPI config so it's clear that the trigger chain is MPSL -> PPI -> EGU -> SWI.

Actually, I believe a comment (either in the setup function or README if you prefer) explaining the trigger chain would be great.

Reply:
L293-5